Assets/MixedRealityToolkit/Utilities/BuildAndDeploy/BuildDeployWindow.cs

```diff
@@ -1,1654 +0,0 @@
-            {
-                currentConnectionInfoIndex = 0;
-            }
-
-            GUILayout.BeginVertical();
-            EditorGUI.BeginChangeCheck();
-            GUILayout.BeginHorizontal();
-
-            GUI.enabled = IsHoloLensConnectedUsb;
-
-            if (GUILayout.Button(pairHoloLensUsbLabel, GUILayout.Width(128f)))
-            {
-                EditorApplication.delayCall += PairDevice;
-            }
-
-            GUI.enabled = true;
-
-            GUILayout.FlexibleSpace();
-
-            var previousLabelWidth = EditorGUIUtility.labelWidth;
-            EditorGUIUtility.labelWidth = 64;
-            bool useSSL = EditorGUILayout.Toggle(useSSLLabel, UwpBuildDeployPreferences.UseSSL);
-            EditorGUIUtility.labelWidth = previousLabelWidth;
-
-            currentConnectionInfoIndex = EditorGUILayout.Popup(currentConnectionInfoIndex, targetIps);
-
-            DeviceInfo currentConnection = portalConnections.Connections[currentConnectionInfoIndex];
-            bool currentConnectionIsLocal = IsLocalConnection(currentConnection);
-
-            if (currentConnectionIsLocal)
-            {
-                currentConnection.MachineName = IsHoloLensConnectedUsb ? HOLOLENS_USB : LOCAL_MACHINE;
-
-                if (currentConnection.MachineName != targetIps[0])
-                {
-                    UpdatePortalConnections();
-                }
-            }
-
-            GUI.enabled = IsValidIpAddress(currentConnection.IP);
-
-            if (GUILayout.Button(addConnectionLabel, GUILayout.Width(20)))
-            {
-                portalConnections.Connections.Add(new DeviceInfo(EMPTY_IP_ADDRESS, currentConnection.User, currentConnection.Password));
-                currentConnectionInfoIndex++;
-                currentConnection = portalConnections.Connections[currentConnectionInfoIndex];
-                UpdatePortalConnections();
-            }
-
-            GUI.enabled = portalConnections.Connections.Count > 1 && currentConnectionInfoIndex != 0;
-
-            if (GUILayout.Button(removeConnectionLabel, GUILayout.Width(20)))
-            {
-                portalConnections.Connections.RemoveAt(currentConnectionInfoIndex);
-                currentConnectionInfoIndex--;
-                currentConnection = portalConnections.Connections[currentConnectionInfoIndex];
-                UpdatePortalConnections();
-            }
-
-            GUI.enabled = true;
-
-            GUILayout.EndHorizontal();
-            GUILayout.Space(5);
-            GUILayout.BeginHorizontal();
-            GUILayout.FlexibleSpace();
-
-            GUILayout.Label(currentConnection.MachineName, GUILayout.Width(HALF_WIDTH));
-
-            GUILayout.EndHorizontal();
-
-            previousLabelWidth = EditorGUIUtility.labelWidth;
-            EditorGUIUtility.labelWidth = 64;
-            GUILayout.BeginHorizontal();
-            GUILayout.FlexibleSpace();
-
-            GUI.enabled = !currentConnectionIsLocal;
```

Comment:
Was this case removed intentionally? Seems reasonable that we wouldn't want people editing the default local case, which shouldn't have much variation from `127.0.0.1` or `Local Machine`.

Reply:
yes it didn't manage the lifecycle of this hardcode addition well. So you could easily break and then it would lock you out. So you would have to delete and re-create.

Reply:
Do you have repro steps for that? I haven't seen it reported before

Reply:
And managing that state better while still preventing the user from overwriting it and getting into a bad state feels preferable. We probably want the local state to always be represented

Reply:
Unfortunately I don't remember all of the items I found in detail because I worked on this over a month ago. 

I remember one was if I typed in 127.0.0.1, it would lock it out and I wouldn't be able to edit it anymore. So you would have to remove it and would make it difficult to add port number.

There was also some weird stuff I think with machine name possibly. 

Basically the code was trying to lock down UI input when it was a local connection...but it wasn't doing that in all code parts.

Reply:
>if I typed in 127.0.0.1

That's the purpose of the default local machine setting, and it auto-adds the port number since that doesn't change for USB deployment:
https://github.com/microsoft/MixedRealityToolkit-Unity/blob/58b51a014478bd3fcd98dd14bcb15d1e7b3cde15/Assets/MixedRealityToolkit/Utilities/WindowsDevicePortal/DevicePortal.cs#L740
I didn't think manually adding 127.0.0.1 would be needed. Did you need to change the port to something specific for some reason?

>Basically the code was trying to lock down UI input when it was a local connection...but it wasn't doing that in all code parts.

I think we should update the other code paths that weren't locking down instead of removing the lock down. This "Local Machine"/"127.0.0.1" setting should be equivalent to the "Device" setting in Visual Studio, which is always there as an option. Maybe it needs a rename to be more clear? Or two settings: one for the actual PC Local Machine and one for a USB-connected HoloLens (like proposed in https://github.com/microsoft/MixedRealityToolkit-Unity/issues/5698), since this single setting does change behavior based on if a HoloLens is attached.

Reply:
Talked offline with @keveleigh 

Going to simplify both code and interface to just use a boolean toggle. Sort of like in VS 

User selects Local target (i.e 127.0.0.1) or Remote target
if Local, then no ips etc found
If Remote, then list of IPs shown

No special handling of 127.0.0.1 checking unless against the toggle boolean flag